commit b4d88a60fe0e126b245583c5003584cf6751e536
Merge: 483a351ed4d4 a2db328b0839
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Thu May 23 13:44:47 2024 -0700

    Merge tag 'block-6.10-20240523' of git://git.kernel.dk/linux
    
    Pull more block updates from Jens Axboe:
     "Followup block updates, mostly due to NVMe being a bit late to the
      party. But nothing major in there, so not a big deal.
    
      In detail, this contains:
    
       - NVMe pull request via Keith:
           - Fabrics connection retries (Daniel, Hannes)
           - Fabrics logging enhancements (Tokunori)
           - RDMA delete optimization (Sagi)
    
       - ublk DMA alignment fix (me)
    
       - null_blk sparse warning fixes (Bart)
    
       - Discard support for brd (Keith)
    
       - blk-cgroup list corruption fixes (Ming)
    
       - blk-cgroup stat propagation fix (Waiman)
    
       - Regression fix for plugging stall with md (Yu)
    
       - Misc fixes or cleanups (David, Jeff, Justin)"
    
    * tag 'block-6.10-20240523' of git://git.kernel.dk/linux: (24 commits)
      null_blk: fix null-ptr-dereference while configuring 'power' and 'submit_queues'
      blk-throttle: remove unused struct 'avg_latency_bucket'
      block: fix lost bio for plug enabled bio based device
      block: t10-pi: add MODULE_DESCRIPTION()
      blk-mq: add helper for checking if one CPU is mapped to specified hctx
      blk-cgroup: Properly propagate the iostat update up the hierarchy
      blk-cgroup: fix list corruption from reorder of WRITE ->lqueued
      blk-cgroup: fix list corruption from resetting io stat
      cdrom: rearrange last_media_change check to avoid unintentional overflow
      nbd: Fix signal handling
      nbd: Remove a local variable from nbd_send_cmd()
      nbd: Improve the documentation of the locking assumptions
      nbd: Remove superfluous casts
      nbd: Use NULL to represent a pointer
      brd: implement discard support
      null_blk: Fix two sparse warnings
      ublk_drv: set DMA alignment mask to 3
      nvme-rdma, nvme-tcp: include max reconnects for reconnect logging
      nvmet-rdma: Avoid o(n^2) loop in delete_ctrl
      nvme: do not retry authentication failures
      ...

diff --cc block/blk-core.c
index ea44b13af9af,dd29d5465af6..82c3ae22d76d
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@@ -618,7 -619,9 +621,9 @@@ static void __submit_bio(struct bio *bi
  	if (unlikely(!blk_crypto_bio_prep(&bio)))
  		return;
  
+ 	blk_start_plug(&plug);
+ 
 -	if (!bio->bi_bdev->bd_has_submit_bio) {
 +	if (!bdev_test_flag(bio->bi_bdev, BD_HAS_SUBMIT_BIO)) {
  		blk_mq_submit_bio(bio);
  	} else if (likely(bio_queue_enter(bio) == 0)) {
  		struct gendisk *disk = bio->bi_bdev->bd_disk;